packages/terra-date-input/src/DateInput.jsx

```diff
@@ -264,6 +266,16 @@ class DateInput extends React.Component {
     }
   }
 
+  /**
+   * Takes a key press from the day and year input, and processes it based on the value of the keycode
+   * Prevents non-numeric characters from being entered in Safari browser.
+   * @param {Object} event Event object generated from the event delegation.
+   */
+  handleKeyPress = event => {
+    const input = event.key;
+    if (!input.match(/^[0-9]+$/) && !event.metaKey) event.preventDefault();
```

Comment:
Does this change work if a non-numeric key is copied/pasted in the input using mouse click menu or the Command-C/V keys instead of entering the non-numeric key directly using the keyboard?  If copy/paste doesn't work, something to try is to move this logic to `handleValueChange` similar to [this](https://github.com/cerner/terra-framework/blob/terra-date-picker%404.65.0/packages/terra-date-picker/src/DateInput.jsx#L149-L151).

Reply:
Talked to Saket to keep the flexing of the input types and how handle preventing character key for Safari. Also need to handle copy and paste characters.